rules/ast/functions.go

```diff
@@ -498,6 +499,43 @@ func derivImpl(timestamp clientmodel.Timestamp, args []Node) interface{} {
 	return resultVector
 }
 
+// === histogram_quantile(k ScalarNode, vector VectorNode) Vector ===
```

Comment:
To be clear, I'd expect:

histogram_quantile by (instance )(.9, rate(histo[5m])) to produce a time series per instance
and histogram_quantile(.9, rate(histo[5m])) to error out due to clashing labels as histogram_quantile(.9, sum by (le)(rate(histo[5m]))) is the correct form.